src/modules/dashboard/components/Dialogs/ControlSafeDialog/TransactionTypesSection/TransferFundsSection.tsx

```diff
@@ -41,67 +42,107 @@ const MSG = defineMessages({
   balancesError: {
     id: `dashboard.ControlSafeDialog.ControlSafeForm.TransferFundsSection.balancesError`,
     defaultMessage:
-      'Unable to fetch the Safe balances. Please check your connection',
+      'Unable to fetch Safe balances. Please check your connection',
+  },
+  noSafeSelectedError: {
+    id: `dashboard.ControlSafeDialog.ControlSafeForm.TransferFundsSection.noSafeSelectedError`,
+    defaultMessage: 'You must first select a safe',
   },
 });
 
 const displayName = `dashboard.ControlSafeDialog.ControlSafeForm.TransferFundsSection`;
 
 export interface TransferFundsProps extends TransactionSectionProps {
-  handleValidation: () => void;
+  savedTokenState: [{}, React.Dispatch<React.SetStateAction<{}>>];
 }
 
 const renderAvatar = (address: Address, item: AnyUser) => (
   <UserAvatar address={address} user={item} size="xs" notSet={false} />
 );
 
+const Loading = () => (
+  <DialogSection>
+    <div className={styles.spinner}>
+      <SpinnerLoader
+        appearance={{ size: 'medium' }}
+        loadingText={MSG.balancesLoading}
+      />
+    </div>
+  </DialogSection>
+);
+
+interface ErrorMessageProps {
+  error: Message;
+}
+
+const ErrorMessage = ({ error }: ErrorMessageProps) => (
+  <DialogSection>
+    <div className={styles.error}>
+      {typeof error === 'string' ? (
+        <span>{error}</span>
+      ) : (
+        <FormattedMessage {...error} />
+      )}
+    </div>
+  </DialogSection>
+);
```

Comment:
Are these leftovers from a rebase? I remember that they were supposed to be extracted for all sections to use, right?

Reply:
I do it in this one https://github.com/JoinColony/colonyDapp/pull/3998/commits/dd38f34fdace72c4cbb6397b1555cb70220aebef#diff-d9b918ffd427089e5685173ef3ff58bfd6e8016420e3fa43120248e5fe19b88c